manifests/system.pp

```diff
@@ -67,11 +56,13 @@
 
   }
   else {
+    ensure_packages(['curl'])
```

Comment:
I was thinking more of other UNIXes.  The metadata.json lists Solaris and AIX as supported (although I've no idea if this is accurate).  The `exec` works for them?

Be careful with `ensure_packages`, it is not as safe as `if ! defined`.  It will only not create a duplicate resource declaration error if the packages are declared with the same parameters.  eg This fails...
```
package {'curl':
  ensure => '7.47.0-1ubuntu2.1',
}
ensure_packages('curl')
```
but this is ok
```
package {'curl':
  ensure => 'present',
}
ensure_packages('curl')
```